huff0/decompress.go

```diff
@@ -908,6 +889,90 @@ func (d *Decoder) Decompress4X(dst, src []byte) ([]byte, error) {
 	return dst, nil
 }
 
+// decompress4x_main_loop is the portable imlementation of
+// Decoder.Decompress4X main loop.
+func decompress4x_main_loop(br0, br1, br2, br3 *bitReaderShifted,
+	peekBits uint8, buf *[4][decompress4x_aux_buffer]byte, tbl []dEntrySingle) (int, bool) {
+
+	const tlSize = 1 << tableLogMax
+	const mask = tlSize - 1
+	table := tbl[:tlSize]
+
+	const bufoff = decompress4x_aux_buffer
+	exhausted := false // if any of bitReaderShifted got exhausted
+	off := 0
```

Comment:
Having this a uint8 is to eliminate the bounds check on `buf[stream][off]`.

Reply:
Changed